src/app/components/auth/reset-password-confirm/reset-password-confirm.component.spec.ts

```diff
@@ -9,33 +9,112 @@ import {AuthService} from '../../../services/auth.service';
 import {WindowService} from '../../../services/window.service';
 import {ApiService} from '../../../services/api.service';
 import {EndpointsService} from '../../../services/endpoints.service';
+import { NotFoundComponent } from '../../not-found/not-found.component';
+import { ActivatedRoute, Router, Routes } from '@angular/router';
+import { Observable } from 'rxjs';
+
+const routes: Routes = [
+  {
+    path: '',
+    component: ResetPasswordConfirmComponent,
+  },
+  {
+    path: 'reset-password/confirm/:user_id/:reset_token',
+    component: ResetPasswordConfirmComponent
+  },
+  {
+    path: '404',
+    component: NotFoundComponent,
+  },
+  {
+    path: '**',
+    redirectTo: '/404',
+    pathMatch: 'full'
+  }
+];
 
 describe('ResetPasswordConfirmComponent', () => {
   let component: ResetPasswordConfirmComponent;
   let fixture: ComponentFixture<ResetPasswordConfirmComponent>;
+  let router: Router;
+  let route: ActivatedRoute;
+  let apiService: ApiService;
 
   beforeEach(async(() => {
     TestBed.configureTestingModule({
-      declarations: [ ResetPasswordConfirmComponent ],
+      declarations: [ ResetPasswordConfirmComponent, NotFoundComponent ],
       providers: [
         GlobalService,
         AuthService,
         WindowService,
         ApiService,
         EndpointsService
       ],
-      imports: [RouterTestingModule, HttpClientModule, FormsModule],
+      imports: [RouterTestingModule.withRoutes(routes), HttpClientModule, FormsModule],
     })
     .compileComponents();
   }));
 
   beforeEach(() => {
+    router = TestBed.get(Router);
+    route = TestBed.get(ActivatedRoute);
+    apiService = TestBed.get(ApiService);
     fixture = TestBed.createComponent(ResetPasswordConfirmComponent);
     component = fixture.componentInstance;
-    fixture.detectChanges();
   });
 
   it('should create', () => {
-    expect(component).toBeTruthy();
+    router.navigate(['/reset-password/confirm', '1', '100']).then(() => {
+      fixture.detectChanges();
+      console.log(router.url);
```

Comment:
Remove all the console logs. (Check at other places as well.)

Reply:
Done